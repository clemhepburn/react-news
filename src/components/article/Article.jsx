import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, description, author, url }) => (
  <article>
    <h1>{title}</h1>
    <p>{description}</p>
    <p>By {author}</p>
    <p><a href={url}>Read more</a></p>
  </article>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
