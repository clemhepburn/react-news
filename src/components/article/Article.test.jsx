import React from 'react';
import { render } from '@testing-library/react';
import Article from './Article';

describe('article component', () => {
  it('should render the article component', () => {
    const article = {
      title: 'bad movie',
      author: 'Roger Ebert',
      url: 'https://rogerebert.com'
    };

    const { asFragment } = render(
      <Article
        title={article.title}
        author={article.author}
        url={article.url} />
    );

    expect(asFragment).toMatchSnapshot();
  })
})