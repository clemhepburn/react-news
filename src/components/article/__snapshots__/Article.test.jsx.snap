// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article component should render the article component 1`] = `[Function]`;
